src/spikeinterface/generation/hybrid_tools.py

```diff
@@ -0,0 +1,518 @@
+from __future__ import annotations
+
+import warnings
+from typing import Literal
+import numpy as np
+
+from ..core.template import Templates
+
+from ..core.generate import (
+    generate_templates,
+    generate_unit_locations,
+    generate_sorting,
+    InjectTemplatesRecording,
+    _ensure_seed,
+)
+from ..core.template_tools import get_template_extremum_channel
+from ..core.job_tools import split_job_kwargs
+from ..postprocessing.unit_localization import compute_monopolar_triangulation
+
+from .drift_tools import (
+    InjectDriftingTemplatesRecording,
+    DriftingTemplates,
+    make_linear_displacement,
+    interpolate_templates,
+    move_dense_templates,
+)
+
+
+def estimate_templates_from_recording(
+    recording, ms_before=2, ms_after=2, sorter_name="spykingcircus2", **run_sorter_kwargs
+):
+    """
+    Get templates from a recording. Internally, SpyKING CIRCUS 2 is used by default
+    with the only twist that the template matching step is not launched. Instead, a Template
+    object is returned based on the results of the clustering. Other sorters can be invoked
+    with the `sorter_name` and `run_sorter_kwargs` parameters.
+
+    Parameters
+    ----------
+    ms_before : float
+        The time before peaks of templates
+    ms_after : float
+        The time after peaks of templates
+    sorter_name : str
+        The sorter to be used in order to get some fast clustering
+    run_sorter_kwargs : dict
+        The parameters to provide to the run_sorter function of spikeinterface
+
+
+    sorter_params: keyword arguments for `spyking_circus2` function
+
+    Returns
+    -------
+    templates: Templates
+        The found templates
+    """
+    from spikeinterface.sorters.runsorter import run_sorter
+    from spikeinterface.core.template import Templates
+    from spikeinterface.core.waveform_tools import estimate_templates
+
+    if sorter_name == "spykingcircus2":
+        if "matching" not in run_sorter_kwargs:
+            run_sorter_kwargs["matching"] = {"method": None}
+
+    sorting = run_sorter(sorter_name, recording, **run_sorter_kwargs)
+
+    from spikeinterface.core.waveform_tools import estimate_templates
+
+    spikes = sorting.to_spike_vector()
+    unit_ids = np.unique(spikes["unit_index"])
+    sampling_frequency = recording.get_sampling_frequency()
+    nbefore = int(ms_before * sampling_frequency / 1000.0)
+    nafter = int(ms_after * sampling_frequency / 1000.0)
+
+    _, job_kwargs = split_job_kwargs(run_sorter_kwargs)
```

Comment:
No. This is sorters specific.
Some sorters do not have job_kwargs.
Lets explicitily define job_kwargs in the signature instead of hacking some nested one.